### ISSUE TITLE
problems with long-living watch

### ISSUE THREAD

**Issue by eranreshef** [tomplus/kubernetes_asyncio#29]
I was trying to follow one of the basic [examples](https://github.com/tomplus/kubernetes_asyncio/blob/master/examples/example2.py) but replacing the `list_namespaces` func with `read_node`. The major difference between the 2 is that `read_node` requires a `name` argument.
The code looks like this:
```
import asyncio

from kubernetes_asyncio import client, config, watch


async def main():
    # Configs can be set in Configuration class directly or using helper
    # utility. If no argument provided, the config will be loaded from
    # default location.
    await config.load_kube_config()

    v1 = client.CoreV1Api()
    count = 20
    w = watch.Watch()
    my_node_name = 'some-node-that-actually-exists'
    async for event in w.stream(v1.read_node, my_node_name):
        print("{} Event: {} {}\n\labels={}\n\n".format(count, event['type'],
                event['object'].metadata.name, event['object'].metadata.labels))
        count -= 1
        if not count:
            w.stop()

    print("Ended.")


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(main())
    loop.close()
```

The run output is:
```
Traceback (most recent call last):
  File "async_test.py", line 28, in <module>
    loop.run_until_complete(main())
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 468, in run_until_complete
    return future.result()
  File "async_test.py", line 16, in main
    async for event in w.stream(v1.read_node, my_node_name):
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/kubernetes_asyncio/watch/watch.py", line 95, in __anext__
    return await self.next()
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/kubernetes_asyncio/watch/watch.py", line 101, in next
    self.resp = await self.func()
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/kubernetes_asyncio/client/api/core_v1_api.py", line 20285, in read_node
    (data) = self.read_node_with_http_info(name, **kwargs)  # noqa: E501
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/kubernetes_asyncio/client/api/core_v1_api.py", line 20318, in read_node_with_http_info
    " to method read_node" % key
TypeError: Got an unexpected keyword argument 'watch' to method read_node
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x1101f16a0>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x11020b240>
```

Is this a bug or am I doing something wrong?

**Comment by tomplus** [comment]
In your case `stream()` adds an extra argument `watch=True` to the method but [read_node](https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/docs/CoreV1Api.md#read_node) doesn't support watching. It's a limit of K8s API - only collections of objects can be watched (for example [list_nodes](https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/docs/CoreV1Api.md#list_node) can be watched).

**Comment by eranreshef** [comment]
Thanks!

**Comment by eranreshef** [comment]
Another question please (since I got your attention).
When I call `list_nodes` without setting a `timeout_seconds`, after a couple of minutes I get:
```Traceback (most recent call last):
  File "async_test.py", line 36, in <module>
    loop.run_until_complete(main())
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 468, in run_until_complete
    return future.result()
  File "async_test.py", line 17, in main
    async for event in w.stream(v1.list_node, timeout_seconds=0):
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/kubernetes_asyncio/watch/watch.py", line 95, in __anext__
    return await self.next()
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/kubernetes_asyncio/watch/watch.py", line 109, in next
    line = await self.resp.content.readline()
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/aiohttp/streams.py", line 282, in readline
    await self._wait('readline')
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/aiohttp/streams.py", line 250, in _wait
    await waiter
  File "/Users/erares01/.virtualenvs/splatt-master/lib/python3.6/site-packages/aiohttp/helpers.py", line 677, in __exit__
    raise asyncio.TimeoutError from None
concurrent.futures._base.TimeoutError
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x10ca99390>```

What am I doing wrong here?

**Comment by tomplus** [comment]
Yes, API server has its own timeout. `Watch` should manage with it. I'll prepare fix for it over the weekend.

**Comment by tomplus** [comment]
@eranreshef fixed, please upgrade to 1.0.0b3. Thanks.

**Comment by eranreshef** [comment]
It seems that now, when calling watch with no timeout value, I stop receiving events after ~20 minutes.
I created one watch call with no timeout value, and a infinite loop of watch calls with `timeout_seconds=30` . The first watch stops receiving events after ~20 minutes and the watch in loop keeps receiving events.
